overlord/snapstate/handlers_components.go

```diff
@@ -381,8 +382,9 @@ func (m *SnapManager) doUnlinkCurrentComponent(t *state.Task, _ *tomb.Tomb) (err
 		return err
 	}
 
-	// set information for undoUnlinkCurrentComponent in the task
-	t.Set("unlinked-component", unlinkedComp)
+	// set information for undoUnlinkCurrentComponent in the change
+	t.Change().Set(fmt.Sprintf("unlinked-component-%s",
```

Comment:
we need to discuss this, attaching things to the change directly like this is very unusual

Reply:
Changed as discussed, also for the kernel previous revision on kernel updates.